lib/mixins/properties-mixin.html

```diff
@@ -0,0 +1,208 @@
+       * listed in `properties`.
+       */
+      static get observedAttributes() {
+        const props = this._properties;
+        return props ? Object.keys(props).map(p => this.attributeNameForProperty(p)) : [];
+      }
+
+      /**
+       * Finalizes an element definition, including ensuring any super classes
+       * are also finalized. This includes ensuring property
+       * accessors exist on the element prototype. This method calls
+       * `_finalizeClass` to finalize each constructor in the prototype chain.
+       */
+       static finalize() {
+        if (!this.hasOwnProperty(JSCompiler_renameProperty('__finalized', this))) {
+          const superCtor = superPropertiesClass(this);
+          if (superCtor) {
+            superCtor.finalize();
+          }
+          this.__finalized = true;
+          this._finalizeClass();
+        }
+      }
+
+      /**
+       * Finalize an element class. This includes ensuring property
+       * accessors exist on the element prototype. This method is called by
+       * `finalize` and finalizes the class constructor.
+       *
+       * @protected
+       */
+      static _finalizeClass() {
+        const props = ownProperties(this);
+        if (props) {
+          this.createProperties(props);
+        }
+      }
+
+      /**
+       * Returns a memoized version of all properties, including those inherited
+       * from super classes. Properties not in object format are converted to
+       * at least {type}.
+       *
+       * @return {Object} Object containing properties for this class
+       * @protected
+       */
+      static get _properties() {
+        if (!this.hasOwnProperty(
+          JSCompiler_renameProperty('__properties', this))) {
+          const superCtor = superPropertiesClass(this);
+          this.__properties = Object.assign({},
+            superCtor && superCtor._properties,
+            ownProperties(this));
+        }
+        return this.__properties;
+      }
+
+      /**
+       * Overrides `PropertiesChanged` method to return type specified in the
+       * static `properties` object for the given property.
+       * @param {string} name Name of property
+       * @return {*} Type to which to deserialize attribute
+       *
+       * @protected
+       */
+      static typeForProperty(name) {
+        const info = this._properties[name];
+        return info && info.type;
+      }
+
+      /**
+       * Overrides `PropertiesChanged` method and adds a call to
+       * `finalize` which lazily configures the element's property accessors.
+       * @override
+       */
+      _initializeProperties() {
+        this.constructor.finalize();
+        super._initializeProperties();
+      }
+
+      /**
+       * Called when the element is added to a document.
+       * Calls `_enableProperties` to turn on property system from
+       * `PropertiesChanged`.
+       */
+      connectedCallback() {
+        this._enableProperties();
```

Comment:
This should check for and call `super.connectedCallback`

Reply:
Confirming whether closure supports the `if (super.connectedCallback)` incantation and will add if so (if not, that will be unfortunate but will document that this needs to be the first mixin applied that implements `connectedCallback`).